client/packages/coldchain/src/Equipment/DetailView/Tabs/Summary.tsx

```diff
@@ -408,9 +399,8 @@ export const Summary = ({ draft, onChange, locations }: SummaryProps) => {
         <Section heading={t('label.donor')}>
           <Row isGaps={isGaps} label={t('label.donor')}>
             <DonorSearchInput
-              value={draft.donor as NameRowFragment} // Using as NameRowFragment is ok, because the comparison function is based on the id
-              onChange={e => onChange({ donor: e, donorNameId: e?.id })}
-              onInputChange={onDonorInputChange}
+              donorId={draft.donor?.id ?? null}
```

Comment:
Updated the DonorSearchInput to only need ID rather than whole name row fragment, and allow setting to null.